python/ray/data/_internal/execution/resource_manager.py

```diff
@@ -0,0 +1,123 @@
+import time
+from typing import TYPE_CHECKING, Dict
+
+import ray
+from ray.data._internal.execution.interfaces.execution_options import (
+    ExecutionOptions,
+    ExecutionResources,
+)
+from ray.data._internal.execution.interfaces.physical_operator import PhysicalOperator
+from ray.data._internal.execution.operators.input_data_buffer import InputDataBuffer
+
+if TYPE_CHECKING:
+    from ray.data._internal.execution.streaming_executor_state import Topology
+
+
+class ResourceManager:
```

Comment:
It's mostly a `ResourceMonitor` / `ResourceTracker`. `ResourceManager` makes me think if they are responsible for scheduling resource (launch/kill tasks).

Reply:
I actually plan to add the per-op resource allocation to this class in the next PR. At that time, monitor or tracker will be inaccurate. Thus I prefer keeping it for now. If we consider it as "managing the resource usages", it still kind of makes sense

Reply:
SG